tests/unit/llms/test_llm_loader.py

```diff
@@ -62,43 +64,44 @@ def test_constructor_no_model():
         LLMLoader(provider=constants.PROVIDER_BAM, model=None)
 
 
-def test_constructor_wrong_provider():
-    """Test how wrong provider is checked."""
-    with pytest.raises(UnsupportedProviderError):
-        LLMLoader(provider="invalid-provider", model=constants.GRANITE_13B_CHAT_V1)
+def test_constructor_unknown_provider():
+    """Test how a provider missing from configuration is checked for."""
+    with pytest.raises(UnknownProviderError):
+        LLMLoader(provider="unknown-provider", model=constants.GRANITE_13B_CHAT_V1)
 
 
 def test_constructor_unsupported_provider():
-    """Test how wrong provider is checked."""
-    provider_config = ProviderConfig()
-    provider_config.models = {constants.GRANITE_13B_CHAT_V1: "mock-model"}
-    config.llm_config.providers = {"unsupported-provider": provider_config}
-    with pytest.raises(UnsupportedProviderError):
-        LLMLoader(provider="unsupported-provider", model=constants.GRANITE_13B_CHAT_V1)
-
+    """Test how a configured but an unsupported provider (not openai, bam, etc) is checked for."""
+    test_provider = "unsupported-provider"
+    config.config = Config(
+        {
+            "llm_providers": [
+                {
+                    "name": test_provider,
+                    "models": [
+                        {
+                            "name": constants.GRANITE_13B_CHAT_V1,
+                        }
+                    ],
+                }
+            ]
+        }
+    )
```

Comment:
this seems harder to read and maintain than the previous code that called the individual constructors was, what is the motivation for the refactor?  (same question/comment for the other spot where you did this)

Reply:
The way config was contructed in this code was breaking (the thing layer of) config encapsulation. For example, I couldn't quite tell what this
```
    provider_config = ProviderConfig()
    provider_config.models = {constants.GRANITE_13B_CHAT_V1: "mock model"}
    config.llm_config.providers = {provider: provider_config}
```
was even doing, `models` is a `dict[str, ModelConfig]`, it was plugging in a string literal instead of a ModelConfig.
Some other cases were impossible according to the config model, such as providers with no models raise an InvalidConfigurationError("no models configured for provider...").